File: kotlin/kotlin-vs-java/java-example/src/main/java/org/example/ex08/Exam08.java
package org.example.ex08;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.Optional;

/**
 * default value
 */
public class Exam08 {

    public Exam08(Store store) {
        String str = toLocalDateTimeString(store.getRegisteredAt());
    }

    public String toLocalDateTimeString(LocalDateTime localDateTime) {

        localDateTime = Optional.ofNullable(localDateTime).orElseGet(LocalDateTime::now);

        return localDateTime.format(DateTimeFormatter.ofPattern("yyyy MM dd"));
    }

    public static void main(String[] args) {
        Store store = new Store();

        new Exam08(store);
    }
}

class Store {

    private LocalDateTime registeredAt;

    public LocalDateTime getRegisteredAt() {
        return registeredAt;
    }

    public void setRegisteredAt(LocalDateTime registeredAt) {
        this.registeredAt = registeredAt;
    }
}

File: kotlin/kotlin-vs-java/kotlin-example/src/main/kotlin/ex08/Exam08.kt
package ex08

import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

/**
 * default value
 */
fun main() {
    Exam08(Store())

    println(DateTimeUtil().localDateTimeToString())
    println(DateTimeUtil().localDateTimeToString(LocalDateTime.now()))
}

data class Store(
    var registeredAt: LocalDateTime? = null
)

class Exam08(store: Store) {

    init {
        println(toLocalDateTimeString(store.registeredAt))
        println(toLocalDateTimeString())
    }

    fun toLocalDateTimeString(
        localDateTime: LocalDateTime? = LocalDateTime.of(2022, 2, 2, 2, 2, 2)
    ): String {
        return (localDateTime ?: LocalDateTime.now()).format(DateTimeFormatter.ofPattern("yyyy MM dd"))
    }
}

class DateTimeUtil {

    private val KST_FORMAT = "yyyy년 MM월 dd일 HH시 mm분 ss초"

    fun localDateTimeToString(
        localDateTime: LocalDateTime? = LocalDateTime.now(),
        pattern: String? = KST_FORMAT
    ): String {
        return (localDateTime?: LocalDateTime.now()).format(DateTimeFormatter.ofPattern(pattern));
    }
}